iceoryx_utils/README.md

```diff
@@ -69,12 +72,27 @@ queue which is thread-safe when combined with `smart_lock`. To learn more about
 |:-----------------------:|:--------:|:--------------:|:------------|
 |`ActiveObject`       | i | X | Active object base skeleton implementation inspired by [Prefer Using Active Objects Instead Of Naked Threads](https://www.drdobbs.com/parallel/prefer-using-active-objects-instead-of-n/225700095)  |
 |`FiFo`               | i |   | Single Producer, Single Consumer Lock Free FiFo |
+|`smart_lock`         | i |   | Creates arbitrary thread safe constructs which then can be used like smart pointers. If some STL type should be thread safe use the smart_lock to create the thread safe version in one line. Based on some ideas presented in [Wrapping C++ Member Function Calls](https://stroustrup.com/wrapper.pdf) |
+|`TriggerQueue`       | i | X | Queue with a `push` - `pop` interface where `pop` is blocking as long as the queue is empty. Can be used as a building block for active objects. |
 |`LockedLoFFLi`       | i | X | LIFO based index manager (Lock Free Free List). One building block of our memory manager. After construction it contains the indices {0 ... n} which you can acquire and release. |
 |`LoFFLi`             | i |   | Lock-free version of LockedLoFFLi. |
-|`smart_lock`         | i |   | Creates arbitrary thread safe constructs which then can be used like smart pointers. If some STL type should be thread safe use the smart_lock to create the thread safe version in one line. Based on some ideas presented in [Wrapping C++ Member Function Calls](https://stroustrup.com/wrapper.pdf) |
 |`SoFi`               | i |   | Single Producer, Single Consumer Lock Free Safely overflowing FiFo (SoFi). |
 |`TACO`               | i |   | Thread Aware exChange Ownership (TACO). Solution if you would like to use `std::atomic` with data types larger than 64 bit. Wait free data synchronization mechanism between threads.|
-|`TriggerQueue`       | i | X | Queue with a `push` - `pop` interface where `pop` is blocking as long as the queue is empty. Can be used as a building block for active objects. |
+
+an atrribute overview 
+
+| Data Structure | Shared Memory usable  | Thread-Safe | Lock-Free | Concurrent Producers : Consumers | Bounded Capacity | Data Type Restriction |Use Case |
+|----------------|-----------------------|-------------|-----------|----------------------------------|------------------|---|-----------------------|
+|`FiFo`          | Yes | No  | No  | n/a | Yes | Copyable            | FIFO Data transfer | 
+|`smart_lock`    | No  | Yes | No  | n/a | n/a | Copyable            | Wrapper to make classes thread-safe (by using a lock)| 
```

Comment:
The `smart_lock` can be used in the shared memory and has no data type restrictions!

Reply:
Fixed!